core/vm/eips.go

```diff
@@ -319,3 +320,110 @@ func enable6780(jt *JumpTable) {
 		maxStack:    maxStack(1, 0),
 	}
 }
+
+func enable3074(jt *JumpTable) {
+	jt[AUTH] = &operation{
+		execute:     opAuth,
+		constantGas: 3100,
+		dynamicGas:  gasAuthEIP2929,
+		minStack:    minStack(3, 1),
+		maxStack:    maxStack(3, 1),
+		memorySize:  memoryAuth,
+	}
+	jt[AUTHCALL] = &operation{
+		execute:     opAuthCall,
+		constantGas: params.CallGasEIP150,
+		dynamicGas:  gasCallEIP2929,
```

Comment:
gasCallEIP2929 relies on warm storage cost to be already included. The constantGas should then have additional params.WarmStorageReadCostEIP2929?

Reply:
Also, gasCallEIP2929 itself uses logic which doesn't satisfy all steps needed for AUTHCALL dynamic gas calculation. For example, when "If the gas operand is equal to 0, the instruction will send all available gas as per [EIP-150](https://eips.ethereum.org/EIPS/eip-150)."
`gasCallEIP2929` doesn't obey this

Reply:
in fact the testcase should ideally fail:
1. GAS opcode is followed by AUTHCALL. GAS pushes the remaining gas (after GAS op exec)
2. for AUTHCALL execution, dynamic gas is calculated and then out of the remaining gas, `all_but_one_64th` is calculated. This would be always less than gas on stack (pushed by GAS above). 


The EIP mentions this logic:

```
if gas == 0:
    subcall_gas = all_but_one_64th
elif all_but_one_64th < gas:
    raise                       # Execution is invalid.
else:
    subcall_gas = gas
```

gas is not 0, and the second case is always satisfied, leading to invalid execution

Reply:
I edit the test, replacing GAS opcode with PUSH0:

https://github.com/ledgerwatch/erigon/pull/10154/files#diff-5fd0ea2f1f3910fe70b219d59a388b920a342bbbf807b89eb6f70543ff0853ddR86

Erigon's draft PR is [here](https://github.com/ledgerwatch/erigon/pull/10154), and includes the gas cost calculation differences mentioned in eip-3074.